DEVELOPING_OPAL/validate/src/test/java/org/opalj/fpcf/fixtures/alias/ShapeTest.java

```diff
@@ -0,0 +1,229 @@
+        shapeTest.getLastOr(new Square(1.0));
+        shapeTest.getLastOrCircle(new Square(1.0));
+    }
+
+    @Alias(mayAlias = {@MayAlias(reason = "param might be added to field", id = "ShapeTest.shapesAddParam")},
+            noAlias = {@NoAlias(reason = "param is never added to field", id = "ShapeTest.shapeGetLastOrParam")})
+    Shape[] shapes = new Shape[10];
+    int index = 0;
+
+    Shape currentShape;
+    
+    @Alias(mayAlias = {@MayAlias(reason = "", id = "ShapeTest.addRectangle")})
+    public Shape addRectangle(
+            @Alias(mayAlias = {@MayAlias(reason = "", id = "ShapeTest.addRectangle"), @MayAlias(reason = "param might be assigned to field", id = "ShapeTest.shapesAddParam")})
+            Shape shape) {
+
+        if (shape instanceof Rectangle) {
+            shapes[index++] = shape;
+            return shapes[index - 1];
+        } else {
+            return null;
+        }
+
+    }
+
+    @Alias(mayAlias = {@MayAlias(reason = "return value might be param", id = "ShapeTest.getLastOrParam"),
+                       @MayAlias(reason = "return value might be null", id = "ShapeTest.getLastOrNull", aliasWithNull = true)})
+    public Shape getLastOr(
+            @Alias(mayAlias = {@MayAlias(reason = "return value might be param", id = "ShapeTest.getLastOrParam")}, 
+                    noAlias = {@NoAlias(reason = "param is never added to field", id = "ShapeTest.shapeGetLastOrParam")})
+            Shape shape) {
+
+        Shape currentShape = shape;
+
+        for (int i = 0; i < shapes.length; i++) {
+
+            Shape next = shapes[i];
+            if (next != null) {
+                currentShape = next;
+            }
+        }
+
+        return currentShape;
+    }
+
+    @Alias(mayAlias = {@MayAlias(reason = "return value might be param", id = "ShapeTest.getLastOrCircleParam"),
+            @MayAlias(reason = "return value might be new circle", id = "ShapeTest.getLastOrCircleNewCircle"),
+            @MayAlias(reason = "return value might be null", id = "ShapeTest.getLastOrNull", aliasWithNull = true)})
+    public Shape getLastOrCircle(
+            @Alias(mayAlias = {@MayAlias(reason = "return value might be param", id = "ShapeTest.getLastOrCircleParam")},
+                   noAlias = {@NoAlias(reason = "param is never new circle", id = "ShapeTest.getLastOrCircleParamNewCircle")}
+            )
+            Shape shape) {
+
+        Shape currentShape = shape;
+
+        Shape circle = new @Alias(mayAlias = {@MayAlias(reason = "return value might be new circle", id = "ShapeTest.getLastOrCircleNewCircle")},
+                                  noAlias = {@NoAlias(reason = "param is never new circle", id = "ShapeTest.getLastOrCircleParamNewCircle")}
+        ) Circle(2.0);
+
+        for (int i = 0; i < shapes.length; i++) {
+
+            Shape next = shapes[i];
+            if (next != null) {
+                currentShape = next;
+            }
+        }
+
+        if (!(currentShape instanceof Circle)) {
```

Comment:
I'm not sure what the point in all of these checks is. Try to write the tests so that for every line that is there, it is clear why it is necessary and what it is supposed to test. Try to separate different things that you want to test into different files with each testing one specific category of things. Name variables, functions, etc. in a way that describes what they test. Rectangles, shapes, adding them, etc. doesn't have a meaning in terms of the test.